### PR TITLE
Minimize and Maximize support

### PR DESCRIPTION
**Pull request by JoseCarlosGarcia95**
Hello,
I'm making my first contribution to Mathics project giving symbolic optimization support.

### CODE REVIEW

**Comment by JoseCarlosGarcia95** [comment]
Sorry for multiple commits in this pull request, it's my first pull request.

**Comment by JoseCarlosGarcia95** [comment]
This function support one variable, multiples variables and constraints

File: mathics/builtin/optimization.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+            'List',
+            from_sympy(objective_function.subs(minimum).simplify()),
+            [Expression('Rule', from_sympy(list(minimum.keys())[i]), from_sympy(list(minimum.values())[i])) for i in range(len(vars_sympy))]
+        ) for minimum in minimum_list))
+    
+class Maximize(Builtin):
+    """
+    <dl>
+    <dt>'Maximize[$f$, $x$]'
+        <dd>compute the minimum of $f$ respect $x$ that change between $a$ and $b$
```
Comment (axkr) [inline mathics/builtin/optimization.py]: compute the maximum

File: mathics/builtin/optimization.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+    #>> Maximize[1 - (x y - 3)^2, {x, y}]
+     = {{1, {x -> 3, y -> 1}}}
+
+    #>> Maximize[{x - 2 y, x^2 + y^2 <= 1}, {x, y}]
+     = {{Sqrt[5], {x -> Sqrt[5] / 5, y -> -2 Sqrt[5] / 5}}}
+    """
+
+    attributes = ('ReadProtected',)
+    simpy_name = 'Maximize'
```
[the remark this answers is not in the document]
Comment (wolfv) [reply mathics/builtin/optimization.py]: or rather: does this sympy function exist? I think the sympy_name attribute might need to map to a sympy funciton

**Comment by wolfv** [comment]
looks very great to me! maybe @poeschko can review this, too?

File: mathics/builtin/optimization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoseCarlosGarcia95) [reply mathics/builtin/optimization.py]: Nope sympy has not any function for calculating minimum or maximum

**Comment by JoseCarlosGarcia95** [comment]
I'm making this using karish kunt tucker conditions :)

File: mathics/builtin/optimization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (poeschko) [reply mathics/builtin/optimization.py]: Perhaps this should be removed, then?

**Comment by wolfv** [comment]
@JoseCarlosGarcia95 does that mean this PR is not yet ready and you're still working on it?

**Comment by JoseCarlosGarcia95** [comment]
Yes is ready for master. works fine at least for me. And it's complete

**Comment by JoseCarlosGarcia95** [comment]
It was a typo with my English ;)

**Comment by wolfv** [comment]
I think you should remove the `simpy_name` and just not set it to anything. then we can merge.

**Comment by JoseCarlosGarcia95** [comment]
Squashed all commits and deleted simpy_name

**Comment by JoseCarlosGarcia95** [comment]
> Squashed all commits and deleted simpy_name @wolfv

After deleting simpy_name, one of travis checks fails (ALL_MODULES=true PYTHON2.7) but it's look like a general issue

**Comment by JoseCarlosGarcia95** [comment]
@poeschko

**Comment by JoseCarlosGarcia95** [comment]
What about this? @poeschko

**Pull request merged**